app/models/ambassador.rb

```diff
@@ -0,0 +1,41 @@
+class Ambassador < SimpleDelegator
+  def self.all
+    decorate(User.ambassadors)
+  end
+
+  def self.decorate(target)
+    unless target.is_a?(User) || target.respond_to?(:all?) && target.all? { |e| e.is_a?(User) }
+      raise ArgumentError, "Must be a User or collection of Users"
+    end
+
+    Array(target).map { |u| new(u) }
+  end
+
+  def initialize(user)
+    unless user.is_a?(User) && user.ambassador?
+      raise ArgumentError, "Must be an ambassador"
+    end
+    super(user)
+  end
+
+  def percent_complete
+    return 0.0 if ambassador_task_assignments.empty?
+    (completed_tasks_count / tasks_count.to_f).round(2)
+  end
+
+  def progress_count
+    "#{completed_tasks_count}/#{tasks_count}"
+  end
+
+  def completed_tasks_count
+    ambassador_task_assignments.select(&:completed?).count
+  end
+
+  def tasks_count
+    ambassador_task_assignments.count
+  end
+
+  def groups
```

Comment:
Why use groups rather than organizations here?

Reply:
I think this might be better named `ambassador_organizations` - and then you could just have the method return `organizations.ambassadors` - which is important